crates/fuel-core/src/service/genesis/importer.rs

```diff
@@ -171,11 +164,7 @@ impl SnapshotImporter {
                 progress_reporter,
             );
             async move {
-                if num_groups >= GROUPS_NUMBER_FOR_PARALLELIZATION {
-                    tokio_rayon::spawn(move || task.run()).await?;
-                } else {
-                    task.run()?;
-                }
+                tokio_rayon::spawn(move || task.run()).await?;
```

Comment:
is this change related to cancellation? are there risks to creating too many spawned rayon tasks here?

Reply:
> is this change related to cancellation?

No. This got introduced at some point (don't know when), I just caught it now.

Callbacks given to `TaskManager::spawn` must be light. A [rule of thumb](https://ryhl.io/blog/async-what-is-blocking/) given by a tokio developer is between 10 and 100 us between awaits.

Allowing any number of batches to execute directly in the callback is problematic since batch size can be arbitrary. The import logic is not async and will not yield until all batches are done.

> are there risks to creating too many spawned rayon tasks here?

It's suboptimal. The rayon thread pool thread number is fixed by default, also low, read somewhere it is 1x core count.

Our workload is a combination of blocking IO (rocksdb) and CPU intensive work (decompression, decoding, merkle calculations etc) probably best fitted to spawning a thread per job since we have a limited amount of tables (jobs) to migrate.

So exhausting the thread pool might provide suboptimal CPU utilization when threads are stuck on blocking IO. Also will introduce a sort of semaphore where whole tables will not begin the import process due to a lack of available threads.

Rayon isn't fit for long running IO-blocking tasks. Should be used for processing batches of CPU intensive work (has support for work-stealing).

 I'll use `tokio::spawn_blocking` for now since the backing thread pool seems to expand based on demand. I'll go back to rayon in a followup PR but will actually sub-divide the work and send batches off to be processed as rayon tasks.

Reply:
I've changed it to use the tokio thread pool for blocking tasks, i.e. `join_set.spawn_blocking`. There are a lot more threads in that pool by default. Will see if that solves the issue. 

I'd like to avoid to do any amount of batches inline since we may prevent the executor from driving other futures. But if spawn_blocking is still slow (i.e. flaky p2p tests in the CI, I'll return the job inlining).